api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -710,20 +713,36 @@ public void mergePatients_shouldMergeVisitsFromNonPreferredToPreferredPatient()
 		assertThat(mergeLogData.getMovedEncounters(), containsInAnyOrder(encounterUuidsThatShouldBeMoved.toArray()));
 	}
 	
-	private ArgumentMatcher<Visit> matchingVisit(final Visit expected) {
-		return new ArgumentMatcher<Visit>() {
-			
-			@Override
-			public boolean matches(Object argument) {
-				Visit visit = (Visit) argument;
-				return OpenmrsUtil.nullSafeEquals(visit.getLocation(), expected.getLocation())
-				        && OpenmrsUtil.nullSafeEquals(visit.getVisitType(), expected.getVisitType())
-				        && OpenmrsUtil.nullSafeEquals(visit.getIndication(), expected.getIndication())
-				        && OpenmrsUtil.nullSafeEquals(visit.getStartDatetime(), expected.getStartDatetime())
-				        && OpenmrsUtil.nullSafeEquals(visit.getStopDatetime(), expected.getStopDatetime())
-				        && (visit.getEncounters().size() == expected.getEncounters().size());
-			}
-		};
+	private Matcher<Visit> matchingVisit(final Visit expected) {
+   return new Matcher<Visit>() {
+	   @Override
+	   public void describeTo(Description description) {
+         
+	   }
+
+	   @Override
+   		public boolean matches(Object argument) {
+		   Visit visit = (Visit) argument;
+		   return OpenmrsUtil.nullSafeEquals(visit.getLocation(), expected.getLocation())
+			   && OpenmrsUtil.nullSafeEquals(visit.getVisitType(), expected.getVisitType())
+			   && OpenmrsUtil.nullSafeEquals(visit.getIndication(), expected.getIndication())
+			   && OpenmrsUtil.nullSafeEquals(visit.getStartDatetime(), expected.getStartDatetime())
+			   && OpenmrsUtil.nullSafeEquals(visit.getStopDatetime(), expected.getStopDatetime())
+			   && (visit.getEncounters().size() == expected.getEncounters().size());
+	   }
+
+	   @Override
+	   public void describeMismatch(Object o, Description description) {
+		   
+	   }
+
+	   @Override
+	   public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {
+		   
```

Comment:
What are these empty methods for?

Reply:
@dkayiwa sir previously we were using hamcrest argument matcher interface (with only one function) but now to avoid the conflict( as argument matcher is used in mockito 2.x)  hamcrest have this matcher interface to use this we have to override all the methods so these are empty functions that are not used.

Reply:
@dkayiwa any update on this or any changes that are needed?

Reply:
@dkayiwa any update on this issue, I have resolved the conflicts in issue and I have checked its working on local